eos/b-decays/b-to-dstar-l-nu.cc

```diff
@@ -93,146 +282,221 @@ namespace eos
+        inline b_to_dstar_l_nu::AngularObservables integrated_angular_observables(const double & s_min, const double & s_max)
         {
-            return normalized_differential_decay_width(s) * tau_B / hbar;
+            return b_to_dstar_l_nu::AngularObservables{ _integrated_angular_observables(s_min, s_max) };
         }
     };
 
-    BToDstarLeptonNeutrino::BToDstarLeptonNeutrino(const Parameters & parameters, const Options & options) :
-        PrivateImplementationPattern<BToDstarLeptonNeutrino>(new Implementation<BToDstarLeptonNeutrino>(parameters, options, *this))
+    BToDstarLeptonNeutrino::BToDstarLeptonNeutrino(const Parameters & p, const Options & o) :
+        PrivateImplementationPattern<BToDstarLeptonNeutrino>(new Implementation<BToDstarLeptonNeutrino>(p, o, *this))
     {
     }
 
     BToDstarLeptonNeutrino::~BToDstarLeptonNeutrino()
     {
     }
 
-    double
-    BToDstarLeptonNeutrino::differential_branching_ratio(const double & s) const
-    {
-        return _imp->differential_branching_ratio(s);
-    }
+    /* q^2-differential observables */
 
+    // |Vcb|=1
     double
-    BToDstarLeptonNeutrino::integrated_branching_ratio(const double & s_min, const double & s_max) const
+    BToDstarLeptonNeutrino::normalized_differential_branching_ratio(const double & s) const
     {
-        std::function<double (const double &)> f = std::bind(&Implementation<BToDstarLeptonNeutrino>::differential_branching_ratio, _imp.get(), std::placeholders::_1);
-        return integrate<GSL::QAGS>(f, s_min, s_max);
+        return _imp->differential_angular_observables(s).normalized_decay_width() * _imp->tau_B / _imp->hbar;
     }
 
-    // normalized_differential_branching_ratio (|V_cb|=1)
     double
-    BToDstarLeptonNeutrino::normalized_differential_branching_ratio(const double & s) const
+    BToDstarLeptonNeutrino::differential_branching_ratio(const double & s) const
     {
-        return _imp->normalized_differential_branching_ratio(s);
+        return _imp->differential_angular_observables(s).normalized_decay_width() * std::norm(_imp->model->ckm_cb()) * _imp->tau_B / _imp->hbar;
     }
 
-    // normalized(|Vcb|=1) integrated branching_ratio including NP
     double
-    BToDstarLeptonNeutrino::normalized_integrated_branching_ratio(const double & s_min, const double & s_max) const
+    BToDstarLeptonNeutrino::differential_a_fb_leptonic(const double & s) const
     {
-        std::function<double (const double &)> f = std::bind(&Implementation<BToDstarLeptonNeutrino>::normalized_differential_branching_ratio, _imp.get(), std::placeholders::_1);
-        return integrate<GSL::QAGS>(f, s_min, s_max);
+        return _imp->differential_angular_observables(s).a_fb_leptonic();
     }
 
     double
-    BToDstarLeptonNeutrino::differential_r_d(const double & s) const
+    BToDstarLeptonNeutrino::differential_ratio_tau_mu(const double &s) const
     {
-        double br_muons;
+        double br_tau;
```

Comment:
Is it sufficient here to save only m_l? Should'nt you also save the opt_l, because
this will decide whether you are using the "b->c mu nu_mu" or the "b->c tau nu_tau"
new physics Wilson coefficients?! Maybe in your test cases you should also use
different NP values for mu and tau Wilson coefficients and not the same.

Reply:
Does SaveOption work with SwitchOption yet?

Reply:
i have no idea on this. @cBobeth If this is needed could you explicitly type me down the line that is to be added below that line? According to numerics that i get from eos, it seems to me that saving only m_l is sufficient and it does distinguish between the Wilson coefficients's flavors too.

Reply:
When the Wilson coefficients are requested by the command:

const WilsonCoefficients<BToC> wc = model->wilson_coefficients_b_to_c(opt_l.value(), false);

then there is explicitly "opt_l" required to specify the lepton-type,
so I wonder how this can be correct if you only save the value
of "m_l" and not the one of "opt_l"?

Reply:
Please see integrated R_Dstar, where it is correct and add the lines

Save<std::string> save_opt_l(_imp->opt_l._value, "mu");

Save<std::string> save_opt_l(_imp->opt_l._value, "tau");

as you have done there

Reply:
i think, I have already another line saving opt_l too  it is just below the line saving m_l;

Save<std::string> save_opt_l(_imp->opt_l._value, "tau");

this is done for the integrated ratio level, but not at the differential ratio level.  So do you simply mean to include this also in the differential level too?

Reply:
okay I included the missing "Save<std::string> save_opt_l(_imp->opt_l._value, "tau");" and "Save<std::string> save_opt_l(_imp->opt_l._value, "mu");" worked. thanks.